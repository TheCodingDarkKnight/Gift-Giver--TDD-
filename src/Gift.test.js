import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Gift from "./Gift";

configure({ adapter: new Adapter() });

describe("Gift", () => {
    const gift = shallow(<Gift />)

    it("Renders properly", () => {
        expect(gift).toMatchSnapshot();
    })

    it("Initializes a person and present in `state`", () => {
        expect(gift.state()).toEqual({ person: '', present: '' });
    })

    describe('When typing into the person input', () => {
        const person = "Uncle";

        beforeEach(() => {
            gift.find(".input-person").simulate("change", { target: { value: person } });
        });

        it("Updates the person in `state`", () => {
            expect(gift.state().person).toEqual(person);
        });
    });

    describe('When typing into the present input', () => {
        const present = "Golf Clubs";

        beforeEach(() => {
            gift.find(".input-present").simulate("change", { target: { value: present } });
        });

        it("Updates the present in `state`", () => {
            expect(gift.state().present).toEqual(present);
        });
    });
})