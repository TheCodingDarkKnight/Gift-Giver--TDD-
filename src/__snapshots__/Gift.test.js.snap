// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Renders properly 1`] = `ShallowWrapper {}`;
